airflow/utils/deprecation_tools.py

```diff
@@ -23,25 +23,70 @@
 from types import ModuleType
 
 
-def getattr_with_deprecation(imports: dict[str, str], module: str, name: str):
+def getattr_with_deprecation(
+    imports: dict[str, str],
+    module: str,
+    override_deprecated_classes: dict[str, str],
+    extra_message: str,
+    name: str,
+):
+    """
+    Retrieve the imported attribute from the redirected module ad raises a deprecation warning.
+
+    :param imports: dict of imports and their redirection for the module
+    :param module: module name
+    :param override_deprecated_classes: override target classes with deprecated ones. If target class is
+       found in the dictionary, it will be displayed in the warning message.
+    :param name: attribute name
```

Comment:
Missing extra_message.

```suggestion
    :param extra_message: extra message to display in the warning or import error message
    :param name: attribute name
```